flyteplugins/go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -280,7 +280,14 @@ func BuildRawPod(ctx context.Context, tCtx pluginsCore.TaskExecutionContext) (*v
 			return nil, nil, "", err
 		}
 
-		primaryContainerName = c.Name
+		// If primaryContainerName is set in taskTemplate config, use it instead
+		// of c.Name
+		if val, ok := taskTemplate.GetConfig()[PrimaryContainerKey]; ok {
+			primaryContainerName = val
+			c.Name = primaryContainerName
```

Comment:
could we add a small unit test for it

Reply:
I am wondering whether this part is needed. As I know, to set the primary container name, we need to define in `pod_template=PodTemplate(primary_container_name="primary")` . But if we set this, we will get into `case *core.TaskTemplate_K8SPod` [here](https://github.com/flyteorg/flyte/blob/70cfdff981f866bf03b3d9cbff5ec709c4b5ef9c/flyteplugins/go/tasks/pluginmachinery/flytek8s/pod_helper.go#L296). Therefore, this `if val, ok := taskTemplate.GetConfig()[PrimaryContainerKey]` will never be True.

I am thinking of removing this and use `TaskTemplate_K8SPod` in `spark_test.go` instead.